build.sh

```diff
@@ -144,6 +145,21 @@ build_coreclr()
     fi
 
     if [ $__SkipConfigure == 0 ]; then
+        # if managed build is suported, then set __GenerateVersionSource to true
+        if [ $__SkipMSCorLib == 0 ]; then __GenerateVersionSource=true; fi
+        # Drop version.c file
+        __versionSourceFile=$__ProjectRoot/bin/obj/version.c
+        __versionCPPFile=$__ProjectRoot/bin/obj/version.cpp
+        if [ $__GenerateVersionSource == true ]; then
+            "$__ProjectRoot/init-tools.sh" > "$__ProjectRoot/init-tools.log"
+            echo "Running: \"$__ProjectRoot/Tools/corerun\" \"$__ProjectRoot/Tools/MSBuild.exe\" \"$__ProjectRoot/build.proj\" /t:GenerateVersionSourceFile /p:NativeVersionSourceFile=$__versionSourceFile /p:GenerateVersionSourceFile=true /v:minimal $__OfficialBuildIdArg"
+            "$__ProjectRoot/Tools/corerun" "$__ProjectRoot/Tools/MSBuild.exe" "$__ProjectRoot/build.proj" /t:GenerateVersionSourceFile /p:NativeVersionSourceFile=$__versionSourceFile /p:GenerateVersionSourceFile=true /v:minimal $__OfficialBuildIdArg
```

Comment:
I agree, the problem is that we want to have the logic in one place for both the managed libraries and the native ones and we want this logic to apply to other repos like corefx. For now, we have decided this logic to be in a msbuild task, but we will try to see if there is a better solution.